src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/NetServer.java

```diff
@@ -235,4 +254,17 @@ public void stop() {
     sendDataHandler = null;
     subscribeHandler = null;
   }
+
+  public void shutdown() {
+    stop();
+    // Actually stop the server
+    if (server != null) {
+      server.shutdown();
+      try {
+        server.awaitTermination(1, TimeUnit.MINUTES);
+      } catch (InterruptedException e) {
+        server.shutdownNow();
```

Comment:
Same comment as above. You need to propagate the interrupted status by calling Thread.currentThread().interrupt();

Reply:
Done